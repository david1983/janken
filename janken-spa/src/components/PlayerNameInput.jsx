import React from 'react';
import PropTypes from 'prop-types';
import { observer, inject } from 'mobx-react';
import { Input } from '@material-ui/core';
import GameClass from '../state/Game';

const PlayerNameInput = ({ Game, playerNumber }) => (
  <div className="player-selection">
    <h3 className="label">
      {`Player ${playerNumber}`}
    </h3>
    {' '}
    <Input
      required
      type="text"
      className="input"
      placeholder={`type name for player ${playerNumber}`}
      onChange={e => Game.setPlayerName(playerNumber, e.target.value)}
    />
  </div>
);

PlayerNameInput.defaultProps = {
  Game: new GameClass(),
  playerNumber: 1,
};

PlayerNameInput.propTypes = {
  Game: PropTypes.object,
  playerNumber: PropTypes.number,
};


export default inject('Game')(observer(PlayerNameInput));
