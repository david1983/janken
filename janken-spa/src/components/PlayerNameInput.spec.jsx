import React from 'react';
import { render } from 'enzyme';
import PlayerNameInput from './PlayerNameInput';
import GameClass from '../state/Game';

const Game = new GameClass();
describe('the PlayerNameInput component', () => {
  it('should match the snapshot of the shallow rendered component', () => {
    const component = <PlayerNameInput Game={Game} />;
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
  });
});
