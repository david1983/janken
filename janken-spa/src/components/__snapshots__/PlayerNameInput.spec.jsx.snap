// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the PlayerNameInput component should match the snapshot of the shallow rendered component 1`] = `
<div
  class="player-selection"
>
  <h3
    class="label"
  >
    Player 1
  </h3>
   
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline input"
    required=""
  >
    <input
      class="MuiInputBase-input MuiInput-input"
      placeholder="type name for player 1"
      required=""
      type="text"
    />
  </div>
</div>
`;
